src/ngx_http_lua_directive.c

```diff
@@ -464,8 +466,27 @@ ngx_http_lua_rewrite_by_lua(ngx_conf_t *cf, ngx_command_t *cmd, void *conf)
         return NGX_CONF_ERROR;
     }
 
-    if (llcf->rewrite_handler) {
-        return "is duplicate";
+    if (p_rewrites == NULL) {
+        p_rewrites = ngx_array_create(cf->pool, 1,
+                                      sizeof(ngx_http_lua_phase_ctx_t));
+
+        if (p_rewrites == NULL) {
+            return NGX_CONF_ERROR;
+        }
+
+        llcf->rewrites = p_rewrites;
+    }
+
+    if (p_rewrites->nelts >= NGX_HTTP_LUA_MAX_PHASE_COUNT) {
+        ngx_conf_log_error(NGX_LOG_ERR, cf, 0,
+                           "max rewrite directives limit to %d",
+                           NGX_HTTP_LUA_MAX_PHASE_COUNT);
+        return NGX_CONF_ERROR;
+    }
+
+    phase_ctx = ngx_array_push(p_rewrites);
+    if (phase_ctx == NULL) {
+        return "not enough memory";
```

Comment:
Again, we do not return such C strings upon allocation failures in the config handlers.